### DISCUSSION TITLE
AMD GPUs now work for iw3 on WSL (Windows Subsystem for Linux)

### DISCUSSION THREAD

**Discussion by beosliege** [nagadomi/nunif#166] [General]
I'm happy to report that my 7900 XTX GPU is working nicely on Windows 11 with iw3. If anyone else wants to use their ([supported](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html)) AMD GPU, I hope this will help you:

First ensure you have WSL up and running. Make sure to use Ubuntu 22.04:
https://learn.microsoft.com/en-us/windows/wsl/install

Then I followed these instructions with the latest GPU driver:
https://rocm.docs.amd.com/_/downloads/radeon/en/latest/pdf/

While PyTorch is not yet supported on AMD / ROCm directly in Windows yet, the Linux subsystem in Windows will now do the trick. Yay, no more dual booting! :)

Pages / options I used from the above instructions:

-Pages 1 through 8: Option A 'Graphics" usecase on page 8.
-I didn't set the group permissions on page 9.
-Page 11, I used option A.
-Bottom of page 12, verify PyTorch install.

I fumbled through the virtual environment setup a little bit for iw3 b/c instead of using the requirements.txt file that comes with iw3 (and thus installing nvidia torch packages which wouldn't work), I instead installed these not only in the base python environment in WSL, but also in the virtual environment for iw3:

wget https://repo.radeon.com/rocm/manylinux/rocm-rel-6.1.3/torch-2.1.2%2Brocm6.1.3-cp310-cp310-linux_x86_64.whl
wget https://repo.radeon.com/rocm/manylinux/rocm-rel-6.1.3/torchvision-0.16.1%2Brocm6.1.3-cp310-cp310-linux_x86_64.whl
wget https://repo.radeon.com/rocm/manylinux/rocm-rel-6.1.3/pytorch_triton_rocm-2.1.0%2Brocm6.1.3.4d510c3a44-cp310-cp310-linux_x86_64.whl
pip3 uninstall torch torchvision pytorch-triton-rocm numpy
pip3 install torch-2.1.2+rocm6.1.3-cp310-cp310-linux_x86_64.whl torchvision-0.16.1+rocm6.1.3-cp310-cp310-linux_x86_64.whl pytorch_triton_rocm-2.1.0+rocm6.1.3.4d510c3a44-cp310-cp310-linux_x86_64.whl numpy==1.26.4

Then I basically just kept trying to launch iw3 and watching it fail on missing packages and kept installing each package until it worked.

Once you get to this text on page 13, iw3 should work:
"Environment set-up is complete, and the system is ready for use with PyTorch to work with machine learning models,
and algorithms"

**Comment by nagadomi** [comment]
Thank you for the report.
If it works on Linux or WSL, I believe it will simply work if AMD supports ROCm Pytorch for Windows.

As for the dependent packages,
We currently have a separate `requirements-torch.txt` file, so I think we just need to create a file like `requirements-torch-rocm.txt` and use that instead. Possibly `numpy` needs to be removed from `requirements.txt`.
